src/components/CCPForm/CCPFormPage.tsx

```diff
@@ -0,0 +1,186 @@
+import { useMutation } from '@apollo/react-hooks';
+import { useQuery } from 'react-apollo';
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { ValidatorForm } from 'react-material-ui-form-validator';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import { Colours } from '../../styles/Constants';
+import CompleteButton from './CompleteButton';
+import Map from '../EventCreationPage/Map';
+import FormField from '../common/FormField';
+import { GET_ALL_CCPS, GET_CCP_BY_ID, CCP } from '../../graphql/queries/ccps';
+import { ADD_CCP, EDIT_CCP } from '../../graphql/mutations/ccps';
+
+const useStyles = makeStyles({
+  ccpWrapper: {
+    backgroundColor: 'white',
+  },
+  ccpFormTopSection: {
+    margin: '48px 30px 0px 30px',
+    backgroundColor: 'white',
+    borderBottom: '1px solid #c4c4c4',
+  },
+  ccpHeader: {
+    display: 'flex',
+    padding: '16px 0px',
+  },
+  ccpForm: {
+    padding: '56px 56px 101px 56px',
+  },
+  ccpCompleteDiv: {
+    marginBottom: 71,
+    marginTop: 31,
+    display: 'flex',
+    justifyContent: 'flex-end',
+  },
+  ccpCancelBtn: {
+    minWidth: '107px',
+    minHeight: '48px',
+    fontSize: '18px',
+    fontWeight: 500,
+    color: Colours.Secondary,
+  },
+  ccpBtnPosition: {
+    display: 'flex',
+    marginLeft: 'auto',
+    alignSelf: 'center',
+  },
+});
+
+const CCPFormPage = ({
+  match: {
+    params: { eventID, ccpID },
+  },
+  mode,
+}: {
+  match: {
+    params: {
+      eventID: string;
+      ccpID?: string;
+    };
+  };
+  mode: string;
+}) => {
+  const classes = useStyles();
+  const history = useHistory();
+
+  const { data } = useQuery(
+    mode === 'new' ? GET_ALL_CCPS : GET_CCP_BY_ID,
+    mode === 'new'
```

Comment:
you don't actually need this ternary here - I think you can pass in variables even if the query doesn't need it. It's up to you if you want this here, but I'd consider taking the second ternary out for clarity

Reply:
Ok, will do!